language_formatters_pre_commit_hooks/pretty_format_yaml.py

```diff
@@ -42,27 +65,38 @@ def pretty_format_yaml(argv=None):
     # Prevent ruamel.yaml to wrap yaml lines
     yaml.width = maxsize
 
+    separator = '---\n'
+
     for yaml_file in set(args.filenames):
         with open(yaml_file) as f:
             string_content = ''.join(f.readlines())
 
-        try:
-            content = yaml.load(string_content)
+        # Split multi-document file into individual documents
+        #
+        # Not using yaml.load_all() because it reformats primitive (non-YAML) content. It removes
+        # newline characters.
+        separator_pattern = '^---\\s*\\n'
+        original_docs = re.split(separator_pattern, string_content, flags=re.MULTILINE)
 
-            if not isinstance(content, (list, dict)):
-                # skip files containing primitive types (unstructured text)
-                continue
+        pretty_docs = []
+
+        try:
+            for doc in original_docs:
+                content = process_single_document(doc, yaml)
+                if content is not None:
+                    pretty_docs.append(content)
```

Comment:
```suggestion
                pretty_docs.append(_ process_single_document(doc, yaml))
```

Reply:
I confirmed that this fails my tests. Not making this change.